test/testautovariables.cpp

```diff
@@ -2528,7 +2528,7 @@ class TestAutoVariables : public TestFixture {
               "    }\n"
               "    f();\n"
               "}\n");
-        ASSERT_EQUALS("[test.cpp:5] -> [test.cpp:4] -> [test.cpp:7]: (error) Using lambda that captures local variable 'b' that is out of scope.\n", errout.str());
+        ASSERT_EQUALS("[test.cpp:5] -> [test.cpp:3] -> [test.cpp:4] -> [test.cpp:7]: (error) Using lambda that captures local variable 'b' that is out of scope.\n", errout.str());
```

Comment:
hmm .. I do not think this particular `test.cpp:3:note:Assuming condition is true` is helpful. That assumption is obvious when looking at the other notes. Can this note be removed somehow in this case and only shown in some other cases when it would be non-obvious?

In Clang they show the full execution path.. and well those paths can be very difficult to follow. Let's not repeat such mistake.

Reply:
It doesn't show all conditions, just the ones that cppcheck doesn't know the value. However, I think I can remove the note for these more obvious cases.

Reply:
> However, I think I can remove the note for these more obvious cases.

ah thanks. I know it's easier said than done to tweak the error messages for all cases.